content/posts/2022-01-29-things-i-learned-in-2021.md

```diff
@@ -0,0 +1,116 @@
+
+## Participated more actively in the open source world
+
+![opensource](/media/2021/opensource.png)
+
+I love open source. I use open source software every-single-day. I use React, Node.js, VS Code and many many other things. Humanity can go that far I think open source plays a big role. The good thing about open source is that everyone can make their lives and others’ lives much easier and more enjoyable. Anyone can contribute to the open source. It does not require you have to write code to core services of a framework or application. It can be anything, from a typo to the recommendations, the feedback about the software. Next year, I target to contribute more to some projects that I like and am interested like msw and Next.js.
+
+## Adopt Typescript
+
+![Typescript](/media/2021/typescript.png)
+
+I worked with Javascript every day for more than 4 years. You might heard “Javascript is great”. Yeah, I love it. You might hear “Javascript is terrible”. Well, I... don’t disagree. I knew the good and the bad things about javascript but I still love it overall. It gives me flexibility, but also gives me “Cannot read properties of undefined” 😂. Until one day, Microsoft introduced Typescript. I heard about it and thought: “Ok, javascript with types, just another tech. No thanks, I’m good with pure javascript”. But a few months ago, I decided to experiment Typescript for a greenfield project at my company. I and my teammates were amazed by how Typescript helps us improve the quality of our code and catch almost all the potential bugs early. Most important, Typescript gives us a high level of confidence to ship our code to production. Typescript also gives Github Copilot a lot of context, so Copilot can be really helpful. I have much fun working with both Typescript and Copilot. Typescript is not just a “type system” as I imagined before. It’s truly your best sidekick.
```

Comment:
`I and my teammates` --> `My teammates and I`

Reply:
`Typescript also gives Github Copilot a lot of context, so Copilot can be really helpful`, hepful for what?